### ISSUE TITLE
KVO signals don't pull initial value

### ISSUE THREAD

**Issue by robertjpayne** [ReactKit/ReactKit#17]
Maybe it's intentional but I'm used to ReactiveCocoa where the KVO signals take the initial value when they are setup.

Thus when you setup the signal it'll fire the signal the first time by pulling the value.

**Comment by inamiy** [comment]
Hi, thank you for feedback.

I found `RACObserve(TARGET, KEYPATH)` is using `NSKeyValueObservingOptionInitial` option 
to pull the first value on subscribe.
But I'm not sure if this is a good **default behavior**
because we often want to concentrate only on new values
rather than considering both initial & new values (not simple).

So instead, I added `KVO.startingSignal()` to achieve `RACObserve`'s behavior.

```
let signal = KVO.startingSignal(obj, "value")	// same as RACObserve()
// ... is equivalent to `let signal = KVO.signal(obj, "value").startWith(obj.value)`
```

Please take a try.

**Comment by robertjpayne** [comment]
Seems reasonable, I think most cases would pull the initial value but this is good!

**Comment by robertjpayne** [comment]
Actually the problem with this approach is that valueForKeyPath isn't necessarily identical to obj.value.

For instance consider that the key path is an optional, if you use obj.value it will be nil while the KVO value will be NSNull.

Would it make more sense to allow the user to specify which KVO options they would like but leave it by default to just changes?

**Comment by inamiy** [comment]
I just saw your #18, and oh yes, I totally forgot that KVO actually returns NSNull for nil value :sweat:
Thanks for pointing out this.

By implementing your fix in #18 and my `KVO.startingSignal()`,
this KVO-signal will no longer send NSNull and `startWith(nil)` just works thus should resolve this issue.

**Comment by inamiy** [comment]
Completed in #19.
